weave-js/src/core/server/remoteHttp.ts

```diff
@@ -352,6 +353,8 @@ export class RemoteHttpServer implements Server {
           additionalHeaders['x-weave-client-cache-key'] = this.clientCacheKey;
         }
 
+        additionalHeaders['x-weave-request-id'] = crypto.randomUUID();
```

Comment:
Recommend we use `x-request-id` rather than inventing our own - less likely to get stripped out by some intermediate system.

Reply:
done